js/tests/gatecrashers_test.mjs

```diff
@@ -0,0 +1,139 @@
+import { once } from 'node:events'
+import * as cp from 'node:child_process'
+import { mkdir, writeFile, chmod } from 'fs/promises'
+
+export const tests = []
+const fetch = _fetch // to redefine the real fetch
+
+const port = 5000
+
+export const setup = async ({}) => {
+  const dir = '.'
+
+  await mkdir(`${dir}/guests`, { recursive: true })
+
+  const createFilesIn = ({ files, dirPath }) =>
+    Promise.all(
+      files.map(([fileName, content]) =>
+        writeFile(`${dirPath}/${fileName}`, JSON.stringify(content)),
+      ),
+    )
+
+  const sendRequest = async (path, options) => {
+    const response = await fetch(`http://localhost:${port}${path}`, options)
+    const { status, statusText, ok } = response
+    const headers = Object.fromEntries(response.headers)
+    let body = ''
+    try {
+      body = await response.json()
+    } catch (err) {
+      body = err
+    }
+    return { status, body, headers }
+  }
+
+  return { tmpPath: dir, createFilesIn, sendRequest }
+}
+
+// Test the server is running and writes the port in stdout
+tests.push(async ({ path, ctx }) => {
+  ctx.server = cp.spawn('node', [`${path}`])
```

Comment:
The server can now be spawned and killed in each test, according to our conversations - in the branch `DEV-3746` you can find a `startServer` function (in the `setup` function) that returns `server` and `message`

Reply:
done